observability/ml-observability.md

```diff
@@ -0,0 +1,44 @@
+# Observability in Machine Learning  
+
+Development process of software system with machine learning component is more complex
+than traditional software as such systems change in three directions: the code, the model and the data.
+We can distinguish 2 stages of such system lifespan: experimentation and production.
+Both stages require specific approach to observability that we will discuss below:
+
+## Model experimentation and tuning
+
+Experimentation is a process of finding suitable machine learning model and its parameters via training and evaluating such models with one or more data sets.
+When developing and tuning machine learning models data scientist or team of data scientists experiment
+with various parameters and models. Team is interested to observe and compare chosen performance metrics and be able to reproduce machine learning model, i.e. re-run training process with given dataset and parameters and produce the same model.
+
+The following tools are available to monitor model development process:
+
+[MLFlow for Databricks](https://docs.microsoft.com/en-us/azure/databricks/applications/mlflow/)
+MLFlow is open source framework that can be hosted on Azure Databricks. You can use MLFlow SDK to log your evaluation metrics or any parameter you would like and track it at experimentation board in Azure Databricks.
+Source code and dataset version are also saved with log snapshot to provide reproducibility.
+
+[Azure Machine Learning Service](https://ml.azure.com/)
+Azure Machine Learning service provides SDK for Python, R and C#  to log your evaluation metrics to AML Experiment. Experiments are viewed in AML dashboard. Reproducibility is achived by storing code or notebook
```

Comment:
Suggest to have Azure Machine Learning service (AML) so that AML can be referenced later